### DISCUSSION TITLE
Move Devices to their own Repository

### DISCUSSION THREAD

**Discussion by zack-vii** [MDSplus/mdsplus#2559] [Ideas]
I already mentioned this in our meeting some time back. We should move the device driver development outof our release branches. these atomic merge/release of unfinished device drivers has side effects on what you can do to improve your driver code.

#2284 should not be ok according to our backwards compatibility rule as it breaks the device node structure, existing trees will not work anymore.

If you dont expect anyone to use your device, why releasing it with mdsplus. If you just do it for an easy distibution on your test site, this is bad practise.

Furthermore we now have them released in stable as well, so we have two versions of the same driver.

I strongly suggest moving the device drivers out of our releases.

**Comment by GabrieleManduchi** [comment]
Hi Timo,
    even if I agree with you on moving device development stuff outside 
main MDSplus branch (we should agree however on how doing it, in order 
to avoid using each a different approach, I would see it as another 
mdsplus project in github, as done for example in MARTe2-MDSplus project 
), I would like to remind that there is a quick way for backward 
compatibility in devices in case some node are removed/added that we use 
routinely, that is to Decompile and Compile again the tree, i.e.

DecompileTree <my_tree>
CompileTree <my_tree>

CompileTree will create new devices instances with the latest version 
and copy the content of other nodes accordingly. Of course this wont 
work is the semantics of some node has changed, must in the vast 
majority of the cases, new nodes are added without affecting the 
semantics of the others.

Gabriele


On 4/15/2021 9:13 AM, Timo Schroeder wrote:
>
> I already mentioned this in our meeting some time back. We should move 
> the device driver development outof our release branches. these atomic 
> merge/release of unfinished device drivers has side effects on what 
> you can do to improve your driver code.
>
> #2284 <https://github.com/MDSplus/mdsplus/pull/2284> should not be ok 
> according to our backwards compatibility rule as it breaks the device 
> node structure, existing trees will not work anymore.
>
> If you dont expect anyone to use your device, why releasing it with 
> mdsplus. If you just do it for an easy distibution on your test site, 
> this is bad practise.
>
> Furthermore we now have them released in stable as well, so we have 
> two versions of the same driver.
>
> I strongly suggest moving the device drivers out of our releases.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/MDSplus/mdsplus/issues/2286>, or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/ACCLESBR7SHPVVVYZSGGZ7LTI2G2RANCNFSM4263ROAQ>.
>

-- 
Gabriele Manduchi

Istituto Gas Ionizzati del CNR
Consorzio RFX - Associazione EURATOM/ENEA sulla Fusione
Corso Stati Uniti 4, 35127 Padova - Italy
ph +39-049-829-5039/-5000 fax +39-049-8700718
***@***.***, http://www.igi.cnr.it
